elastic/apm-server
# apmpackage: use package spec v3

## Motivation/summary
use package spec v3

blockers:
- [x] https://github.com/elastic/apm-server/pull/12028
- [x] Release of package-spec [v3.0.1](https://github.com/elastic/package-spec/releases/tag/v3.0.1)
- [x] https://github.com/elastic/kibana/pull/169981
- [x] https://github.com/elastic/ingest-dev/issues/2572
- [x] https://github.com/elastic/package-spec/pull/650
- [x] https://github.com/elastic/kibana/pull/168842
- [x] https://github.com/elastic/package-spec/issues/644

## Checklist




For functional changes, consider:
- Is it observable through the addition of either **logging** or **metrics**?
- Is its use being published in **telemetry** to enable product improvement?
- Have system tests been added to avoid regression?

## How to test these changes


### Testing compatibility
1. Spin up apm-server and stack by using `tilt up`, makesure apmpackage installation was successful
2. Generate some traces and send data to apm-server, preferably using a demo app instrumented with apm agents
3. Observe whether the documents are indexed correctly by searching documents in Discover.  Also, check if there are any differences in documents and make sure all APM UIs are intact without any issues etc. 
4. 
### Testing upgrade scenario
1. Go to Elastic Cloud, create a deployment with stack version prior to 8.13
2. Generate some traces and send data to apm-server, preferably using a demo app instrumented with apm agents.
3. Perform an upgrade to 8.13, repeat step 2, then observe whether the documents are indexed correctly by searching documents in Discover.  Also, check if there are any differences in documents and make sure all APM UIs are intact without any issues etc. 

## Related issues
closes https://github.com/elastic/apm-server/issues/11643
